### PR TITLE
Add TextOptions.LineSpacing.

### LINKED ISSUE #114: Forward (to be introduced) text line height/leading parameter to Fonts
To control the parameter we are about to add in SixLabors/Fonts#152, we need to add it to `TextGraphicsOptions` and forward it within `DrawTextProcessor<T>`.

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp.Drawing/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Add the `LineSpacing` property to match `RendererOptions`. Fixes #114

### CODE REVIEW

File: src/ImageSharp.Drawing/ImageSharp.Drawing.csproj [hunk 1]
```diff
@@ -20,7 +20,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="SixLabors.Fonts" Version="1.0.0-beta0013"/>
+    <PackageReference Include="SixLabors.Fonts" Version="1.0.0-beta.13.4" />
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/ImageSharp.Drawing.csproj:23]: Our version numbering tags for fonts are messed up. That should be `1.0.0-beta13.4`.

File: src/ImageSharp.Drawing/Processing/TextOptions.cs [hunk 2]
```diff
@@ -103,6 +94,21 @@ public float DpiY
...
+        /// Gets or sets the line spacing. Applied as a multiple of the line height.
+        /// Defaults to 1.
+        /// </summary>
+        public float LineSpacing
+        {
+            get => this.lineSpacing;
+
+            set
+            {
+                Guard.IsTrue(value != 0, nameof(this.LineSpacing), "Value must not be equal to 0.");
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Processing/TextOptions.cs:107]: `0` can lead to an attempt to divide by 0 in Fonts.

**Review by antonfirsov — COMMENTED** [review]
We need regression tests for the feature before merging. I can help adding them, let's say by Wednesday.

Otherwise LGTM.

**Comment by JimBobSquarePants** [comment]
> We need regression tests for the feature before merging. I can help adding them, let's say by Wednesday.
> 
> Otherwise LGTM.

If someone can give me pointers I can get the test written today.

File: src/ImageSharp.Drawing/ImageSharp.Drawing.csproj (hunk 1 above)
Comment (JimBobSquarePants) [reply src/ImageSharp.Drawing/ImageSharp.Drawing.csproj]: I've resolved this by creating a tag (no release to NuGet) of `v1.0.0-beta13.5`. This ensures that everyone receives the correct upgrade from MyGet

**Comment by antonfirsov** [comment]
@JimBobSquarePants I would add a [Theory] `FontShapesAreRenderedCorrectly_WithLineSpacing(TestImageProvider<TPixel> provider, float lineSpacing, int numOfLines)` similar to this one:
https://github.com/SixLabors/ImageSharp.Drawing/blob/ab51be964359c17928580be3acf735216825f497/tests/ImageSharp.Drawing.Tests/Drawing/Text/DrawTextOnImageTests.cs#L167-L205


With `lineSpacing=1,1.5,2`, `numOfLines=2,3,5` probably.

**Comment by JimBobSquarePants** [comment]
@antonfirsov Thanks! Added the tests. All looks good.

**Line Spacing 1.5, Line Count 3**
FontShapesAreRenderedCorrectly_WithLineSpacing_linespacing_1 5_linecount_3

**Line Spacing 1, Line Count 5**
FontShapesAreRenderedCorrectly_WithLineSpacing_linespacing_1_linecount_5

**Line Spacing 2, Line Count 2**
FontShapesAreRenderedCorrectly_WithLineSpacing_linespacing_2_linecount_2

File: tests/ImageSharp.Drawing.Tests/Drawing/Text/DrawTextOnImageTests.cs [hunk 3]
```diff
@@ -201,6 +201,51 @@ public void OpenSansJWithNoneZeroShouldntExtendPastGlyphe<TPixel>(TestImageProvi
...
+        public void FontShapesAreRenderedCorrectly_WithLineSpacing<TPixel>(
+            TestImageProvider<TPixel> provider,
+            float lineSpacing,
+            int lineCount)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            Font font = CreateFont("OpenSans-Regular.ttf", 16);
+
+            var sb = new StringBuilder();
+            string str = "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Maecenas porttitor congue massa. Fusce posuere, magna sed pulvinar ultricies, purus lectus malesuada libero, sit amet commodo magna eros quis urna.";
```
Comment (antonfirsov) [inline tests/ImageSharp.Drawing.Tests/Drawing/Text/DrawTextOnImageTests.cs:220]:
```suggestion
            string str = "We want ... a SHRUBBERY!";
```

By `lineSpacing` I actually meant the total number of lines (text rows), to stress the corner cases of 1 and 2 lines.

**Review by antonfirsov — DISMISSED** [review]
LGTM with one suggestion.

File: tests/ImageSharp.Drawing.Tests/Drawing/Text/DrawTextOnImageTests.cs (hunk 3 above)
Comment (JimBobSquarePants) [reply tests/ImageSharp.Drawing.Tests/Drawing/Text/DrawTextOnImageTests.cs:220]: Hold on, I don't follow here... I thought we were testing to ensure that multiple individual text blocks wrap appropriately and have the correct leading space?

`lineSpacing` repesents the `LineSpacing` line height multiplier. `lineCount` represents the number of lines to render separated by a new line character.

Comment (antonfirsov) [reply tests/ImageSharp.Drawing.Tests/Drawing/Text/DrawTextOnImageTests.cs:220]: Sorry mistyped:

> By ~lineSpacing~ `lineCount` I actually meant the total number of lines (text rows), to stress the corner cases of 1 and 2 lines.~

Comment (JimBobSquarePants) [reply tests/ImageSharp.Drawing.Tests/Drawing/Text/DrawTextOnImageTests.cs:220]: Ah... So you want to see a test without wrapping text? I'm not sure we'll gain much insight there. 

I'd have to measure the text against the bounds of the output in order to gauge wrapping correctly. That's more complicated.

Comment (antonfirsov) [reply tests/ImageSharp.Drawing.Tests/Drawing/Text/DrawTextOnImageTests.cs:220]: It's not about the wrapping. I thought rendering 1, 2 and *a few* lines is enough to test the feature. It seemed irrelevant whether it's manual break or automatic wrapping.

Checking `TextLayout.cs` again, looks like automatic wrapping is relevant here, and we need to test that as well.

I leave this to your judgment.

Comment (JimBobSquarePants) [reply tests/ImageSharp.Drawing.Tests/Drawing/Text/DrawTextOnImageTests.cs:220]: Yeah a manual break resets the location edge of the line and affects glyph location properties. That's why I'm explicitly setting a wrapping width in the tests since it will simply be cutoff otherwise.

https://github.com/SixLabors/Fonts/blob/2974546fb30dfdb4e18b8443c8af10d1353c246d/src/SixLabors.Fonts/TextLayout.cs#L225-L247

The leading line height of the first line is also reset to undo the scaling. 

https://github.com/SixLabors/Fonts/blob/2974546fb30dfdb4e18b8443c8af10d1353c246d/src/SixLabors.Fonts/TextLayout.cs#L123-L136

I'm inclined to leave the test for now. I think text layout could do with some optimization love so we can revisit the tests then.
